drivers/input/pinnacle/pinnacle.c

```diff
@@ -0,0 +1,316 @@
+			break;
+		}
+		k_msleep(50);
+	}
+
+	/* Clear SW_CC after Power on Reset */
+	if (pinnacle_write_spi(spi, PINNACLE_REG_STATUS1, 0x00)) {
+		LOG_ERR("Failed to clear SW_CC in Status1");
+		return -EIO;
+	}
+
+	ret = pinnacle_write_spi(spi, PINNACLE_REG_SYS_CONFIG1, 0x00);
+	if (ret) {
+		LOG_ERR("Failed to write SysConfig1");
+		return ret;
+	}
+	ret = pinnacle_write_spi(spi, PINNACLE_REG_FEED_CONFIG2,
+				 (PINNACLE_FEED_CONFIG2_GLIDE_EXTEND_DISABLE |
+				  PINNACLE_FEED_CONFIG2_SCROLL_DISABLE |
+				  PINNACLE_FEED_CONFIG2_SECONDARY_TAP_DISABLE |
+				  PINNACLE_FEED_CONFIG2_ALL_TAPS_DISABLE));
+	if (ret) {
+		LOG_ERR("Failed to write FeedConfig2");
+		return ret;
+	}
+
+	/* Enable feed */
+	ret = pinnacle_write_spi(
+		spi, PINNACLE_REG_FEED_CONFIG1,
+		(PINNACLE_FEED_CONFIG1_DATA_MODE_ABSOLUTE | PINNACLE_FEED_CONFIG1_FEED_ENABLE));
+	if (ret) {
+		LOG_ERR("Failed to enable Feed in FeedConfig1");
+		return ret;
+	}
+
+	/* Configure count of Z-Idle packets */
+	ret = pinnacle_write_spi(spi, PINNACLE_REG_Z_IDLE, config->idle_packets_count);
+	if (ret) {
+		LOG_ERR("Failed to disable Z-Idle packets");
+		return ret;
+	}
+
+	ret = pinnacle_init_interrupt(dev);
+	if (ret) {
+		LOG_ERR("Failed to initialize interrupts");
+		return ret;
+	}
+
+	return 0;
+}
+
+#define PINNACLE_INTERRUPT_CFG(inst)                                                               \
+	.dr_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, data_ready_gpios, {}),
+
+#define PINNACLE_CLIPPING_CFG(inst)                                                                \
+	.clipping_enabled = DT_INST_PROP(inst, clipping_enable),                                   \
+	.active_range = {                                                                          \
+		.x_min = DT_INST_PROP(inst, active_range_x_min),                                   \
+		.x_max = DT_INST_PROP(inst, active_range_x_max),                                   \
+		.y_min = DT_INST_PROP(inst, active_range_y_min),                                   \
+		.y_max = DT_INST_PROP(inst, active_range_y_max),                                   \
+	},
+
+#define PINNACLE_SCALE_CFG(inst)                                                                   \
+	.scaling_enabled = DT_INST_PROP(inst, scaling_enable),                                     \
+	.resolution = {                                                                            \
+		.x = DT_INST_PROP(inst, scaling_x_resolution),                                     \
+		.y = DT_INST_PROP(inst, scaling_y_resolution),                                     \
+	},
+
+#define PINNACLE_DEFINE_CONFIG(inst)                                                               \
+	static const struct pinnacle_config pinnacle_config_##inst = {                             \
+		.spi = SPI_DT_SPEC_INST_GET(inst,                                                  \
+					    SPI_OP_MODE_MASTER | SPI_MODE_CPHA | SPI_WORD_SET(8) | \
+						    SPI_TRANSFER_MSB,                              \
```

Comment:
use a separate define for the SPI mode, save a nasty line wrap

Reply:
Done